testcases/open_posix_testsuite/conformance/interfaces/free/1-1.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *
+ *	The function will cause the space pointed to by ptr to be deallocated.
+ *
+ * method:
+ *	-Set rlimit equals to 1024 * 1024 * 2.
+ *	-Allocate memory to pointer_1 of size 1024 * 1024 *1.
+ *	-Alloacte memory for pointer_2 of size 1024 *1024 *1.
+ *	But it will fail due to insufficient memory.
+ *	-Try allocating memory to pointer_2 after freeing memory
+ *	allocated to pointer_1.
+ *	-Check whether memory is allocated by malloc().
+ *
+*/
+
+#define _XOPEN_SOURCE 700
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/resource.h>
+#include "posixtest.h"
+
+#define TNAME "free/1-1.c"
+
+int main(void)
+{
+	int i = 0;
+	struct rlimit rlim;
+	char *ptr1 = NULL;
+	char *ptr2 = NULL;
```

Comment:
What is the point of initializing variables that are never used before they are assigned.

Also for return values the most common name is ret not i.

Reply:
Updated in new patch.